api/types/container/hostconfig.go

```diff
@@ -436,8 +435,8 @@ type HostConfig struct {
 	Annotations     map[string]string `json:",omitempty"` // Arbitrary non-identifying metadata attached to container and provided to the runtime
 
 	// Applicable to UNIX platforms
-	CapAdd          strslice.StrSlice // List of kernel capabilities to add to the container
-	CapDrop         strslice.StrSlice // List of kernel capabilities to remove from the container
+	CapAdd          []string          // List of kernel capabilities to add to the container
+	CapDrop         []string          // List of kernel capabilities to remove from the container
```

Comment:
Even if we do want to continue supporting the old format; these two definitely look wrong, as these fields never supported a string (instead of a slice) AFAIK.

Reply:
Looks like I was wrong there; Hmm, so CapAdd/CapDrop;

- switched to using StrSlice in https://github.com/moby/moby/commit/17999c70c39e14a474e69ce10de05a66258e2a5b
- had some fixes related to parsing "string" instead of "slice"; https://github.com/moby/moby/commit/10a3061c5f342477703e96d9b8b2e877d9cdddf5
- looks indeed like they _did_ accept a string instead of a slice before docker 1.7; https://github.com/moby/moby/issues/14318
- Which would be API v1.19 and before; https://github.com/moby/moby/blob/10a3061c5f342477703e96d9b8b2e877d9cdddf5/api/common.go#L20